### PR TITLE
Add heat map data types

### PR DESCRIPTION
**Pull request by hig4342**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/navermaps/index.d.ts [hunk 1]
```diff
@@ -1234,12 +1235,12 @@ declare namespace naver.maps {
         class HeatMap {
             constructor(heatMapOptions?: HeatMapOptions);
             getColorMap(): SpectrumStyle;
-            getData(): LatLng[] | PointArrayLiteral[];
+            getData(): LatLng[] | PointArrayLiteral[] | WeightedLocation[];
```
Comment (SuhwanCha) [inline types/navermaps/index.d.ts:1238]: Only LatLng[] and PointArrayLiteral[] can be return type on this method.

image

File: types/navermaps/index.d.ts [hunk 2]
```diff
@@ -1234,12 +1235,12 @@ declare namespace naver.maps {
...
             getMap(): Map | null;
             getOptions(key?: string): HeatMapOptions;
             redraw(): void;
             setColorMap(colormap: SpectrumStyle, inReverse: boolean): void;
-            setData(data: LatLng[] | PointArrayLiteral[]): void;
+            setData(data: LatLng[] | PointArrayLiteral[] | WeightedLocation[]): void;
```
Comment (SuhwanCha) [inline types/navermaps/index.d.ts:1243]: same here
image

**Review by SuhwanCha — CHANGES_REQUESTED** [review]
There are differences with document and your codes, is there another reference?

File: types/navermaps/index.d.ts (hunk 2 above)
Comment (hig4342) [reply types/navermaps/index.d.ts:1243]: Perhaps the document needs to be changed as well.

Comment (hig4342) [reply types/navermaps/index.d.ts:1243]: loged Debugging
`console.log(heatMap.getData())`
스크린샷 2022-07-22 오후 2 13 29

Comment (SuhwanCha) [reply types/navermaps/index.d.ts:1243]: Does `setData(WeightedLocation[])` work too?

Comment (hig4342) [reply types/navermaps/index.d.ts:1243]: yes.
There is no problem using 'heatMap.setData()' by temporarily changing the type in my project
스크린샷 2022-07-22 오후 2 51 53
.

**Review by SuhwanCha — APPROVED** [review]

File: types/navermaps/index.d.ts (hunk 2 above)
Comment (hig4342) [reply types/navermaps/index.d.ts:1243]: 스크린샷 2022-07-22 오후 3 07 28

**Comment by hig4342** [comment]
Ready to merge

**Pull request merged**